### PR TITLE
Fix //read crash and rename RStack argument "spacing" to "offset"

### PR DESCRIPTION
**Pull request by orbyfied**
that

Edit: also renamed `//read` argument `spacing` to `offset`

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -117,8 +117,8 @@ The amount of times to stack your selection, defaults to 1.
 #### **direction**
 The direction to stack your selection in, defaults to me (the direction you're looking in).
 
-#### **spacing**
-The space between copies, defaults to 2.
+#### **offset**
+The offset between copies, defaults to 2.
```
Comment (Matthias1590) [inline DOCUMENTATION.md:121]: defaults to 1

**Review by Matthias1590 — CHANGES_REQUESTED** [review]

File: DOCUMENTATION.md (hunk 1 above)
Comment (orbyfied) [reply DOCUMENTATION.md:121]: oh wait is the functionality also different?

Comment (orbyfied) [reply DOCUMENTATION.md:121]: im pretty sure it already worked as offset

Comment (Matthias1590) [reply DOCUMENTATION.md:121]: offset is inclusive, my bad, the default in code should be 2

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ Minimizes your selection by removing surrounding layers consisting of only air.
 
 <br>
 
-### `//read [spacing=1] [onBlock=redstone_lamp[lit=true]] [toBase=10] [reverseBits=false]`
+### `//read [offset=1] [onBlock=redstone_lamp[lit=true]] [toBase=10] [reverseBits=false]`
```
Comment (Matthias1590) [inline DOCUMENTATION.md:135]: offset should default to 2

File: desktop.ini [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[.ShellClassInfo]
```
Comment (Matthias1590) [inline desktop.ini:1]: ???

File: src/main/java/tools/redstone/redstonetools/features/commands/BinaryBlockReadFeature.java [hunk 4]
```diff
@@ -21,7 +21,7 @@ public class BinaryBlockReadFeature extends CommandFeature {
             Blocks.REDSTONE_LAMP.getDefaultState().with(RedstoneLampBlock.LIT, true), null, null
     );
 
-    public static final Argument<Integer> spacing = Argument
+    public static final Argument<Integer> offset = Argument
             .ofType(integer(0))
             .withDefault(1);
```
Comment (Matthias1590) [inline src/main/java/tools/redstone/redstonetools/features/commands/BinaryBlockReadFeature.java]: should default to 2

File: src/main/java/tools/redstone/redstonetools/features/commands/BinaryBlockReadFeature.java [hunk 5]
```diff
@@ -50,7 +49,13 @@ protected Feedback execute(ServerCommandSource source) throws CommandSyntaxExcep
...
         var pos2 = boundingBox.getPos2();
         var direction = pos2.subtract(pos1).normalize();
-        var spacingVector = direction.multiply(spacing.getValue() + 1);
+
+        // prevent infinite loop
+        if (direction.lengthSq() == 0) {
+            direction = BlockVector3.at(1, 1, 1);
+        }
+
+        var spacingVector = direction.multiply(offset.getValue() + 1);
```
Comment (Matthias1590) [inline src/main/java/tools/redstone/redstonetools/features/commands/BinaryBlockReadFeature.java:58]: + 1 should be removed

File: src/main/java/tools/redstone/redstonetools/features/commands/BinaryBlockReadFeature.java [hunk 6]
```diff
@@ -21,7 +21,7 @@ public class BinaryBlockReadFeature extends CommandFeature {
             Blocks.REDSTONE_LAMP.getDefaultState().with(RedstoneLampBlock.LIT, true), null, null
     );
 
-    public static final Argument<Integer> spacing = Argument
+    public static final Argument<Integer> offset = Argument
             .ofType(integer(0))
```
Comment (Matthias1590) [inline src/main/java/tools/redstone/redstonetools/features/commands/BinaryBlockReadFeature.java:25]: minimum should be 1

File: desktop.ini (hunk 3 above)
Comment (orbyfied) [reply desktop.ini:1]: tf

**Review by Matthias1590 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 7]
```diff
@@ -32,3 +32,6 @@ bin/
 
 run/
 logs/
+
```
Comment (Matthias1590) [inline .gitignore:35]: I dont think we need this, desktop.ini should just not be commited

Comment (orbyfied) [reply .gitignore:35]: bruh but i do `git add *` all the time

Comment (orbyfied) [reply .gitignore:35]: isnt it useful so people dont accidentally commit it

File: DOCUMENTATION.md [hunk 8]
```diff
@@ -132,7 +132,7 @@ Minimizes your selection by removing surrounding layers consisting of only air.
 
 <br>
 
-### `//read [spacing=1] [onBlock=redstone_lamp[lit=true]] [toBase=10] [reverseBits=false]`
+### `//read [offset=2] [onBlock=redstone_lamp[lit=true]] [toBase=10] [reverseBits=false]`
 Interprets your selection as a binary number and outputs it.
 
 #### **spacing**
```
Comment (Matthias1590) [inline DOCUMENTATION.md:138]: should be renamed to offset

**Review by Matthias1590 — APPROVED** [review]

**Pull request merged**